lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -1437,6 +1438,29 @@ estimateTxCost pp skeleton =
         let LinearFee LinearFunction {..} = getFeePolicy $ txParameters pp
         in Coin $ ceiling $ intercept + slope * fromIntegral size
 
+-- | Calculate the cost of increasing a CBOR-encoded Coin-value by another Coin
+-- with the lovelace/byte cost given by the 'FeePolicy'.
+--
+-- Outputs values in the range of [0, 8 * perByteFee]
+--
+-- >>> let p = FeePolicy (Quantity 0) (Quantity 44)
+--
+-- >>> costOfIncreasingCoin p 4294967295 1
+-- Coin 176 -- (9 bytes - 5 bytes) * 44 lovelace/byte
+--
+-- >>> costOfIncreasingCoin p 0 4294967296
+-- Coin 352 -- 8 bytes * 44 lovelace/byte
+costOfIncreasingCoin
+    :: FeePolicy
+    -> Coin -- ^ Original coin
+    -> Coin -- ^ Increment
+    -> Coin
+costOfIncreasingCoin (LinearFee fee) from delta =
+    costOfCoin (from <> delta) `Coin.difference` costOfCoin from
+  where
+    perByte = round $ slope fee
```

Comment:
Suggestion: would using `ceiling` here be more appropriate?

I'm concerned that `round` might lead to underestimation. (But perhaps I'm wrong.)

```suggestion
    perByte = ceiling $ slope fee
```

Reply:
I guess the reason we use a `Double` (we should change this) is that the Byron fee policy coefficient was `Rational`, and not `Natural`. I was here more worried about a `Natural` somehow getting an extra `0.00000000001` — not sure if that's possible — and being incorrectly rounded up.

But I see that we already use [ceiling](https://github.com/input-output-hk/cardano-wallet/blob/db9c4ada4184619c6ff73344d07ffee5e5a44602/lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs#L1395) so I will change.